consensus/ethash/sealer.go

```diff
@@ -18,18 +18,26 @@ package ethash
 
 import (
 	crand "crypto/rand"
+	"errors"
 	"math"
 	"math/big"
 	"math/rand"
 	"runtime"
 	"sync"
+	"time"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/consensus"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
 )
 
+var (
+	errNoMineWork        = errors.New("No mining work available yet, don't panic.")
+	errInvalidSealResult = errors.New("Invalid or stale proof-of-work solution.")
+	errUndefinedRemoteOp = errors.New("Undefined remote mining operation.")
```

Comment:
Errors need to be composable. I.e. don't terminate them with `.` and don't capitalize them.

Reply:
You won't need `errUndefinedRemoteOp` if you go down the multiple task channel approach.